include/setimmediate.h

```diff
@@ -7,6 +7,10 @@
 #include <memory>
 #include <get-uv-event-loop-napi.h>
 
+#if !defined(NAPI_VERSION) || NAPI_VERSION < 2
+#include <get-symbol-from-current-process.h>
```

Comment:
I think it would be better to always include this file and not fall back to using `uv_handle_get_data` as a fixed symbol that’s linked to the addon, since that hasn’t always been available.

Reply:
Does it means that `get-uv-event-loop-napi-h` also should be changed.
Sorry if question is silly, I'm not deeply familiar with v8/node.

Reply:
@fanatid It’s not a silly question :) The `#ifdef` in `get-uv-event-loop-napi.h` is because `napi_get_uv_event_loop` is only included in the headers since N-API version 2. It would be okay to only use the `get_symbol_from_current_process` variant there, but if we know that the addon is being compiled for N-API version > 2 anyway, we might as well statically load it.

With `uv_handle_set_data()`, the situation is different, because there’s nothing like `NAPI_VERSION` for libuv versions; i.e. we don’t know which versions of libuv the addon targets, so we can’t automatically assume that it’s available.

Reply:
Thanks for explanation. I checked my PR with node v12.10.0 and though that it will works everywhere, what is looks wrong.

Reply:
I think it wouldn’t work when compiling with a recent Node.js version but then running with one that doesn’t have `uv_handle_set_data()` (I’m not sure what versions those are)